File: graphs JAVA/Graph.java
package graphs;

import java.util.*;

class Edge{
	int y;
	int weight;
	
	Edge(int y, int weight){
		this.y = y;
		this.weight = weight;
	}
	
	public String toString() {
		return String.valueOf(y);
	}
}

public class Graph {
	
	int vertices;
	int degree[];
	boolean directed;
	int edges;
	ArrayList<LinkedList<Edge>> adjacencyList;

	public Graph(int vertices, boolean directed) {
		this.vertices = vertices;
		this.degree = new int[vertices + 1];
		this.directed = directed;
		this.edges = 0;
		this.adjacencyList = new ArrayList<LinkedList<Edge>>();

		for (int i = 0; i <= vertices; i++)
			this.adjacencyList.add(new LinkedList<Edge>());
	}
	
	public void insertEdge(int x, int y) {
		insertEdge(x, y, 1);
	}

	public void insertEdge(int x, int y, int weight) {
		adjacencyList.get(x).add(new Edge(y, weight));
		degree[x]++;

		if (!directed) {
			adjacencyList.get(y).add(new Edge(x, weight));
			degree[y]++;
		}

		edges++;
	}

	public void printGraph() {
		for (int i = 1; i <= vertices; i++) {
			Iterator<Edge> it = adjacencyList.get(i).iterator();
			
			System.out.print(i);
			while (it.hasNext())
				System.out.print("->" + it.next());
			System.out.println();
		}
	}
	
	public void displayDegree() {
		for (int i = 0; i < degree.length; i++) {
			System.out.println("Degree of node " + i + ": " + degree[i]);
		}
	}


}

File: data-structures/Graph.mjs
class Edge {
  constructor(y, weight) {
    this.y = y;
    this.weight = weight;
    this.next = null;
  }

  add(e) {
    this.next = e;
  }

  hasNext() {
    return this.next !== null;
  }
}

export class Graph {
  constructor(vertices, isDirected) {
    this.vertices = vertices;
    this.degree = new Array(vertices + 1);
    this.isDirected = isDirected;
    this.edges = 0;
    this.adjacencyList = [];
  }

  insertEdges(edges) {
    edges.forEach((edge) => this.insertEdge(edge[0], edge[1], edge[2]));
  }

  insertEdge(x, y, weight = 1) {
    this.addEdge(x, y, weight);

    if (!this.isDirected) {
      this.addEdge(y, x, weight);
    }

    this.edges++;
  }

  addEdge(vertex1, vertex2, weight) {
    const list = this.adjacencyList[vertex1];
    const e = new Edge(vertex2, weight);

    if (list) {
      e.next = list;
    }
    this.adjacencyList[vertex1] = e;
    this.degree[vertex1]++;
  }

  printGraph() {
    for (let i = 1; i <= this.vertices; i++) {
      let edge = this.adjacencyList[i];

      let str = i;
      while (edge) {
        str += "->" + edge.y;
        edge = edge.next;
      }
      return str;
    }
  }

  displayDegree() {
    for (let i = 0; i < this.degree.length; i++) {
      console.log("Degree of node " + i + ": " + this.degree[i]);
    }
  }
}
